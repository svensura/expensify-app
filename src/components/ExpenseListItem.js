import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment'
import numeral from 'numeraljs'




const ExpenseListItem = (props) => (  
    <div>
        <Link to={`/edit/${props.id}`}>
            <h3>
                {props.description}
            </h3>
        </Link>
        <p>
            {numeral(props.amount / 100).format('$0.0,')} - {moment(props.createdAt).format('DD MMMM YYYY')}
        </p>

    </div>
)


export default ExpenseListItem;