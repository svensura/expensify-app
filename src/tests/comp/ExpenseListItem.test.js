import React from 'react';
import { shallow } from 'enzyme';

import ExpenseListItem from '../../comp/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('SHOULD RENDER <ExpenseListItem /> WITH expenses[0]', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});