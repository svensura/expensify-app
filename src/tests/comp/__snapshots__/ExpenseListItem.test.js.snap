// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHOULD RENDER <ExpenseListItem /> WITH expenses[0] 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      MacBook Pro 13
    </h3>
  </Link>
  <p>
    6000
  </p>
  <p>
    0
  </p>
</div>
`;
